### DISCUSSION TITLE
Storage

### DISCUSSION THREAD

**Discussion by Arinji2** [pocketbase/pocketbase#5267] [Q&A] [answered]
Ok so i have a collection with a string for a word, string for its definition, updated at and created at columns. So now i automatically delete the rows after a while but now i need to save the words for around like forever xD. So would it be smarter to make a different collection for the words which i am saving. Or i can just delete the definition part of the record in the normal collection and make it like an empty string. What takes up less storage?

**Comment by PaulDotterer** [comment]
While I don't really understand exactly what you want to achieve, you could just add a "deleted" flag to every entry. Would certainly take up less space than a completely new collection. But why do you delete them in the first place?

**Reply by Arinji2** [reply]
Well the site is about like figuring out if a word and its given definition is real or not, so every hour i have a cronjob to delete 100 words and add  new words to keep the list fresh. Now this was before i made the ui. While making the ui i figured i should probs show which word you got wrong like a summary.. so before i just deleted the words and their definitions, but now i need to like keep it in track right. 

I was thinking of having an expired flag... but wont stuff like the definition take up a lot of extra storage?
Like 20 new words are added per hour... in a month thats 1k+ words and their definitions in storage

**Reply by PaulDotterer** [reply]
I think I get what your site does. So it's like a game where you have to guess what is real or not? (Very basic assumption)

In order to keep track of what user guessed what word right you would need to keep the data anyways. So you could not really deduplicate the data to that extend. Are you actually running into storage limitations or is this just a case of premature optimization?

**Reply by Arinji2** [reply, marked as answer]
Ok so I figured out a way, basically I keep the actual word functionality seperate from analytics saving. So i have a different collection for just saving words. And another collection to save all the games which have the records to the saved words collection.

This way I can make my words collection only for the actual game and a different collection for saving the words as needed.

**Reply by pnmcosta** [reply]
Just a heads up, not necessarily for this issue/solution, but more around auditing of deleted records.

Instead of a "deleted" flag I prefer to use a "deleted" datetime column instead, this way you can actually traceback when it was deleted and not just if it was deleted. Compounded with efficient logging you can also trace which event caused the delete without having a full auditing feature built.

Checking if a datetime column is set or not is pretty trivial in both SQL and Go lang `types.DateTime{}.IsZero()`.
